image-extension.md

```diff
@@ -0,0 +1,234 @@
+
+### Process
+
+The extension phase MUST be invoked separately for each base image (build or run) that is extended.
+
+**GIVEN:**
+- The final ordered group of Dockerfiles generated during the build-ext phase,
+- A list of build args for each Dockerfile specified during the build-ext phase,
+
+For each Dockerfile in the group in order, the lifecycle MUST apply the Dockerfile to the base image as follows:
+
+- The lifecycle MUST provide each Dockerfile with:
+- A `base_image` build arg
+    - For the first Dockerfile, the value MUST be the original base image.
+    - When there are multiple Dockerfiles, the value MUST be the intermediate image generated from the application of the previous Dockerfile.
+- A `build_id` build arg
+    - The value MUST be a UUID
+- `user_id` and `group_id` build args
+    - For the first Dockerfile, the values MUST be the original `uid` and `gid` from the `User` field of the config for the original base image.
+    - When there are multiple Dockerfiles, the values MUST be the `uid` and `gid` from the `User` field of the config for the intermediate image generated from the application of the previous Dockerfile.
+
+#### Ability to rebase
+
+When extending the run image, a Dockerfile MAY indicate that it preserves the ability to rebase by including the following instruction:
+
+```bash
+LABEL io.buildpacks.rebasable=true
+```
+
+For the final image to be rebasable, the provided run image and all applied Dockerfiles MUST indicate rebasability.
+
+#### Software-Bill-of-Materials
+
+Image extensions MAY write Software Bill of Materials (SBOM) files as described in [Software-Bill-of-Materials](#software-bill-of-materials). The platform MAY generate additional SBOM files describing the extended base image after all Dockerfiles have been applied.
+
+## Data Format
+
+### Files
+
+### extension.toml (TOML)
+
+This section describes the 'Extension descriptor'.
+
+```toml
+api = "<buildpack API version>"
+
+[extension]
+id = "<extension ID>"
+name = "<extension name>"
+version = "<extension version>"
+homepage = "<extension homepage>"
+description = "<extension description>"
+keywords = [ "<string>" ]
+sbom-formats = [ "<string>" ]
+
+[[extension.licenses]]
+type = "<string>"
+uri = "<uri>"
+```
+
+Image extension authors MUST choose a globally unique ID, for example: "io.buildpacks.apt".
+
+The image extension `id`, `version`, `api`, `licenses`, and `sbom-formats` entries MUST follow the requirements defined in the [Buildpack Interface Specification](buildpack.md).
+
+### launch.toml (TOML)
+
+This section describes the `launch.toml` output by image extensions; for buildpacks see the [Buildpack Interface Specification](buildpack.md).
+
+```toml
+[[args]]
+name = "<build arg name>"
+value = "<build arg value>"
+
+[[labels]]
+key = "<label key>"
+value = "<label valu>"
```

Comment:
What is the benefit of exposing another way of setting labels for extensions when they can directly use the dockerfile LABEL instruction?

Reply:
I can't think of any... maybe we should take it out?